src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState } from "react";
+import UserContext from "./Contexts/UserContext";
+import Login from "./Pages/Login";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import AllRecipes from "./Pages/AllRecipes";
+import NotFound from "./Pages/NotFound";
+
+function App() {
+  const [currentUserId, setUserId] = useState(0);
+
+  return (
+    <div className="flex flex-col min-h-screen bg-background">
+      <UserContext.Provider
+        value={{ userId: currentUserId, changeUserId: setUserId }}
+      >
+        <BrowserRouter>
+          <Routes>
+            <Route path="/" element={<Login></Login>} />
```

Comment:
You can terminate these tags with a short syntax if you want. This is neater so is conventional in React for non-containing tags and components.

```suggestion
            <Route path="/" element={<Login />} />
```

Reply:
Done